### ISSUE TITLE
Consider not dropping `None`s

### ISSUE THREAD

**Issue by mferrera** [equinor/fmu-dataio#817]
When generating metadata we have from the beginning dropped all `None`s

https://github.com/equinor/fmu-dataio/blob/a26bac75797d0a2f06a5854e8c145505da2285d1/src/fmu/dataio/dataio.py#L883

When adding new fields to the schema they have been added as `Optional` such that existing metadata without these fields would "validate" against the new schema. They do, but _only_ because the schema will give them a new default value if they are not there. It is not adding a truly optional field, but adding a _required_ field that provides a default value.

With this default value existing metadata has been able to validate against updated schemas.

However, when we drop Nones when exporting metadata we are causing this in a way to happen -- metadata _looks_ "backward compatible" by not containing the optional field, but actually the Pydantic model is dictating it to be a required field.

So TL;DR the metadata being generated by the model is invalid against the schema it represents because it doesn't contain a required field, but we get around that by providing a default value. This works, but is in some sense wrong: if the the field is required it should write metadata with that field.

This would in a way be a breaking change to start writing these `None`s as JSON `null`s, but is more technically correct.
